### PR TITLE
refactor Page components, add Database page CTA

### LINKED ISSUE #391: Refactor `Page` components to follow Website 2.0 design
## Implementation Model

This need a rework on how the Page components work. We need to separate `PageContent` to `HomePageContent` which is used specifically in the homepage. We also need to modify how the Card component works in the sense that 

## Tasks

- [x] Add `Container` component to design system
- [x] Separate container from the page components
- [x] Separate `PageContent` and `HomePageContent`
- [x] Create `HomePageSection` component

### LINKED ISSUE #406: Forward `kebutuhan` query params from `ProvincesPage` to `ProvincePage`
## Description

After the users click on one of the `Kebutuhan` buttons on the home page, we need to persist the information when they click on any province.

## Additional Information

Please read [this Slack thread](https://kawalcovid19.slack.com/archives/C027861JD1T/p1626947302079900) for more context.

### PR DESCRIPTION
**Pull request by resir014**
Closes #391
Closes #406 
Partial work for #333

## Description

Reworking how the Page components work to better implement the Website 2.0 designs, as well as initial steps for the homepage CTA boxes.

image

## Tasks

- [x] Database CTA
- [x] Database menus
- [x] Sembako CTA

### CODE REVIEW

**Review by zainfathoni — CHANGES_REQUESTED** [review]
Overall, it looks good to me.

However, there's a missing piece that we need to implement along with this refactoring.
Without the #406 issue implemented, we cannot persist the information of the `Kebutuhan` that our users click. Hence there will be a missing gap in the user experience.

Please implement #406 within this PR as well, thanks, @resir014!

**Comment by resir014** [comment]
@zainfathoni Done. Need you help debugging the issue I told you earlier on DM.

**Comment by zainfathoni** [comment]
Mbak @ekamuktia, could you please help us to fix this issue in this branch? You can try to reproduce it using the deployment preview link.
We can't merge this PR unless this issue is resolved, because we're risking our user experience here.

https://user-images.githubusercontent.com/6315466/126916739-b404c183-650c-4e91-99dd-82792feabb26.mov

**Comment by ekamuktia** [comment]
@zainfathoni Please check #433

File: __tests__/pages/provinces/index.test.tsx
Comment (zainfathoni) [inline __tests__/pages/provinces/index.test.tsx:21]: Please remove this manual mock with `next-router-mock`, thanks!

**Review by zainfathoni — CHANGES_REQUESTED** [review]

File: components/home/__tests__/homepage-start.test.tsx
Comment (zainfathoni) [inline components/home/__tests__/homepage-start.test.tsx:251]: If we want to keep this snapshot test once we migrate the content to Netlify CMS, we need to remember to mock the values, not relying on the `_content` directory which can be changed by our editors.

File: __tests__/pages/provinces/index.test.tsx
Comment (zainfathoni) [inline __tests__/pages/provinces/index.test.tsx:20]: Is replacing this with `next-router-mock` breaking your test?
If not, please replace it, so we don't need to maintain the mocking code too much. 😁

**Review by zainfathoni — APPROVED** [review]
Approved with some minor notes. ✅

Thanks for working on it, @resir014! 🙏

**Review by zainfathoni — APPROVED** [review]
LGTM 💯

**Pull request merged**
